### PR TITLE
Separate notifications into namespaces by URL

### PR DESCRIPTION
**Pull request by drj11**
The new types `Message` and `Listener` have been created which add a `Suffix` to what was previous a bare `[]byte` and a `chan []byte` respectively.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -57,22 +58,32 @@ func ActionMain(c *cli.Context) {
 	}
 }
 
+type Message struct {
+	Suffix string
+	Body   []byte
+}
+
+type Listener struct {
+	Suffix string
```
Comment (pwaller) [inline main.go]: s/Suffix/Topic/g

File: main.go [hunk 2]
```diff
@@ -181,18 +194,31 @@ func (h *Hookbot) KeyChecker(wrapped http.Handler) http.HandlerFunc {
...
 		log.Printf("Error serving %v: %v", r.URL, err)
 		return
 	}
-	h.message <- message
+
+	url := r.URL
+	// The suffix is everything after the "/notify/"
+	p := strings.SplitN(url.Path, "/", 3)
+	suffix := p[2]
+	log.Printf("suffix: [[[%v]]]", suffix)
```
Comment (pwaller) [inline main.go]: please remove the log printf and (optionally) add a more "proper-looking" log message instead. (I'd be okay with hookbot logging messages it got to its stderr).

Comment (pwaller) [inline main.go]: (by "messages it got" I mean, "topics which have been notified")

File: main.go [hunk 3]
```diff
@@ -181,18 +194,31 @@ func (h *Hookbot) KeyChecker(wrapped http.Handler) http.HandlerFunc {
...
 }
 
 func (h *Hookbot) ServeSubscribe(conn *websocket.Conn) {
 
-	notifications := h.Add()
-	defer h.Del(notifications)
+	url := conn.Request().URL
+	p := strings.SplitN(url.Path, "/", 3)
+	suffix := p[2]
+	log.Printf("websocket suffix: [[[%v]]]", suffix)
```
Comment (pwaller) [inline main.go]: Another log

File: main.go [hunk 4]
```diff
@@ -181,18 +194,31 @@ func (h *Hookbot) KeyChecker(wrapped http.Handler) http.HandlerFunc {
...
+	h.message <- Message{Suffix: suffix, Body: body}
 }
 
 func (h *Hookbot) ServeSubscribe(conn *websocket.Conn) {
 
-	notifications := h.Add()
-	defer h.Del(notifications)
+	url := conn.Request().URL
+	p := strings.SplitN(url.Path, "/", 3)
+	suffix := p[2]
```
Comment (pwaller) [inline main.go]: I might be tempted to use `regexp` here instead..

**Comment by drj11** [comment]
I introduced the `Topic` instead of `Suffix`, and used a `Regexp` which seems safer than `SplitN`

**Pull request merged**